gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +762,809 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
+                    /*
+                    The standard_parallel_1 defines where the local radius is calculated
+                    not the center of Y Cartesian system (which is latitude_of_origin)
+                    But FITS WCS only supports projections on the sphere
+                    we assume here that the local radius is the one computed at the projection center
+                    */
+                        oSRS.SetEquirectangular2(crval2, crval1, crval2, falseEast, falseNorth);
+
+                    /* Lambert Azimuthal Equal Area / ZEA projection */
+                    } else if( strcmp(pstr,"ZEA" ) == 0 ) {
+                        projName.assign("Lambert_Azimuthal_Equal_Area_");
+                        oSRS.SetLAEA(crval2, crval1, falseEast, falseNorth);
+
+                    /* Lambert Conformal Conic 1SP / COO projection */
+                    } else if( strcmp(pstr,"COO" ) == 0 ) {
+                        projName.assign("Lambert_Conformal_Conic_1SP_");
+                        oSRS.SetLCC1SP (crval2, crval1, scale, falseEast, falseNorth);
+
+                    /* Orthographic / SIN projection */
+                    } else if( strcmp(pstr,"SIN" ) == 0 ) {
+                        projName.assign("Orthographic_");
+                        oSRS.SetOrthographic(crval2, crval1, falseEast, falseNorth);
+
+                    /* Point Perspective / AZP projection */
+                    } else if( strcmp(pstr,"AZP" ) == 0 ) {
+                        projName.assign("perspective_point_height_");
+                        oSRS.SetProjection(SRS_PP_PERSPECTIVE_POINT_HEIGHT);
+                        /* # appears to need height... maybe center lon/lat */
+
+                    /* Polar Stereographic / STG projection */
+                    } else if( strcmp(pstr,"STG" ) == 0 ) {
+                        projName.assign("Polar_Stereographic_");
+                        oSRS.SetStereographic(crval2, crval1, scale, falseEast, falseNorth);
+                    } else {
+                        CPLError(CE_Failure, CPLE_AppDefined,
+                             "Unknown projection.");
+                    }
+
+                    projName.append(target);
+                    oSRS.SetProjParm(SRS_PP_FALSE_EASTING,0.0);
+                    oSRS.SetProjParm(SRS_PP_FALSE_NORTHING,0.0);
+
+                    oSRS.SetNode("PROJCS",projName.c_str());
+
+                    oSRS.SetGeogCS(GeogName.c_str(), DatumName.c_str(), target, aRadius, invFlattening,
+                        "Reference_Meridian", 0.0, "degree", 0.0174532925199433);
+                }  else {
+                    CPLError(CE_Failure, CPLE_AppDefined,
+                         "Unknown projection.");
+                }
+            }
+        } else {
+            CPLError(CE_Failure, CPLE_AppDefined,
+                 "No CTYPE keywords: no geospatial information available.");
+            status = 0;
+        }
+    }
+
+    if( m_bLoadPam )
```

Comment:
this looks like inappropriate copy & paste from the GTiff driver. You don't read from PAM (.aux.xml) at all, but read the scale & offset values from FITS keys. So this method should be renamed from `LoadGeoreferencingAndPamIfNeeded` to `LoadGeoreferenceAndScaleOffset` or something along that line. And you don't need two distinct flags m_bLoadPam and m_bReadGeoTransform but just one. And actually you probably don't need any at all. Instead of calling this method from several other methods in a lazy way, you could systematically call it from the Open() method. This would simplify code and make it more robust. The GTiff driver implements lazy loading for performance reason as it is a critical driver, but FITS doesn't need that level of complexity.